collect_app/src/main/java/org/odk/collect/android/utilities/ClientHttpConnection.java

```diff
@@ -65,7 +65,7 @@
     }
 
     @Override
-    public InputStream getInputStream(@NonNull URI uri, String contentType, final int connectionTimeout, Map<String,String> responseHeadsers) throws Exception {
+    public InputStream getInputStream(@NonNull URI uri, String contentType, final int connectionTimeout, Map<String,String> responseHeaders) throws Exception {
```

Comment:
At this point, this method makes the request, gets all the stuff out of the response and returns the InputStream. If we want the cookies, we deal with the `cookieStore()` and if we need the headers, we're proposing to have an output arg with a `Map<String,String>`.

I'd strongly suggest to avoid the output arg, which is often considered a bad practice: it goes against the intuitive understanding that args of a method represent its input, and that the only output of a method is what's declared as its output type in the method's signature.

I propose we return an object with the InputStream and the headers map. 
  - This would make the output of the method much more explicit
  - Any postprocessing of these things (the InputStream and the headers) could happen inside the object, which would improve the design by encapsulating those concerns inside the object instead of having to do it elsewhere. It would make it easier to test as well.
  - We could even expand this object with the response cookies, to break dependency on the concept of a global cookie store, which could not hold with the replacement library we choose.

Reply:
Nevermind! I see you've done that in other commits :)

Reply:
Thanks, the code did evolve into that solution :)